fix(folder): update to work with radiance-folder 3.0.0

mostaphaRoudsari opened the pull request:
see here for the updated folder structure: https://github.com/ladybug-tools/radiance-folder-structure/releases/tag/v3.0.0

BREAKING CHANGE: This commit ends the support for previous versions of Radiance folder.

chriswmackey commented on honeybee_radiance_folder/folder.py:
```diff
@@ -228,425 +354,184 @@ def daylight_matrix_files(self, aperture, receiver=None):
+            folder_type: An integer between 0-2.
+                * 0: grid-based
+                * 1: view-based
+                * 2: includes both views and grids
```
I am guessing that you want this input exposed on the `Model.to.rad_folder` method. Correct?

mostaphaRoudsari replied:
Hmm. Didn't think about it like that but you are right. It should be accessible via the command. Let's have default set to 2 so it creates both folders.

chriswmackey commented on honeybee_radiance_folder/folder.py:
```diff
@@ -228,425 +354,184 @@ def daylight_matrix_files(self, aperture, receiver=None):
+                value. You can pre-defined from ``config`` module. Default is a
+                grid-based ray-tracing folder.
             overwrite (bool): Set to True to overwrite the folder is it already exist.
-        """
```
Should be:
`overwrite (bool): Set to True to overwrite the folder is it already exists. (Default: False).`

FYI, this option to overwrite was not working when I tried to use it in honeybee-radiance. I had to use the `nukedir` method from ladybug in order to be able to overwrite things. So I would recommend writing a test for this.

mostaphaRoudsari replied:
Changed it to nukedir

chriswmackey commented on honeybee_radiance_folder/folder.py:
```diff
@@ -228,425 +354,184 @@ def daylight_matrix_files(self, aperture, receiver=None):
-        """Write model folder.
+        root_folder = self.model_folder(full=True)
+        if overwrite:
+            shutil.rmtree(root_folder, ignore_errors=True)
```
I think this was the method that wasn't working in most of my tests. It might be a good idea to take a look at what happens in the ladybug `nukedir` method to see some more reliable options.

chriswmackey commented on honeybee_radiance_folder/folder.py:
```diff
@@ -228,425 +354,184 @@ def daylight_matrix_files(self, aperture, receiver=None):
                 raise ValueError('{} already exist.'.format(directory))
-            os.makedirs(directory)
 
+        shutil.copy2(self._config_file, root_folder)
```
Interesting. Is `copy2` better than `copy`?

mostaphaRoudsari replied:
The difference is that the second argument can be a folder instead of a file.

chriswmackey replied:
Good to know.

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -8,17 +8,17 @@ class NonApertureState(object):
     """
-    def __init__(self, name, default, direct):
-        self.name = name
+    def __init__(self, identifier, default, direct):
```
Should the name of this class be changed to `SceneState` since anything that uses this class is a part of the `scene_dynamic` folder, right?

chriswmackey replied:
Or, in honeybee-radiance we call this a `ShadeState`. I realize that you might want to use a different naming convention from honeybee since this is all supposed to be specific to Radiance and not the conventions of the Honeybee Model . But I think you should just either follow the own convention you set up here (calling is `SceneState`) or follow the honeybee-core convention (calling it a `ShadeState`). Right now, this name doesn't seem to com from anywhere except an obsolete naming convention.

chriswmackey replied:
Also, if you go with `SceneState`,  it might be good to say in the docstring that this object parallels the honeybee-radiance `ShadeState` just to make it clear to people familiar with honeybee-radiance that these refer to the same thing.

mostaphaRoudsari replied:
I was just lazy in changing it. This is the old name based on the older folder structure. I will do SceneState to match the folder itself and add a comment about shadeState.

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -29,33 +29,38 @@ def from_dict(cls, input_dict):
 
     Args:
-        name (str): Optional human-readable name for the state. Can be None.
+        identifier (str): Optional human-readable identifier for the state. Can be None.
```
It might be good to say in the docstring that this object parallels the honeybee-radiance `SubFaceState` just to make it clear to people familiar with honeybee-radiance that these classes refer to the same thing.

chriswmackey replied:
Or just let me know if you think honeybee-radiance should be changed. I called it `SubFaceState` because it can be assigned to both `Apertures` and `Doors`.

mostaphaRoudsari replied:
I think you picked the right name for inside honeybee-library but for the folder I think ApertureState is more appropriate. I will add the note as you suggested.

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -118,104 +125,103 @@ def from_dict(cls, input_dict):
     @classmethod
     def from_dict(cls, input_dict):
-        """Create a dynamic aperture from a dictionary.
+        """Create a aperture group from a dictionary.
```
This is a `dynamic scene` not an `aperture group`. Also you should probably put some `Args:`

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -84,11 +90,11 @@ def __init__(self, name, default, direct,
     def from_dict(cls, input_dict):
         """Create a state from an input dictionary.
-        
+
```
You should probably put some `Args:` here.

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -118,104 +125,103 @@ def from_dict(cls, input_dict):
+        return cls(identifier, default, direct, black, tmtx, vmtx, dmtx)
 
-    def validate(self, folder):
+    def validate(self, folder, bsdf_folder):
```
You should probably put some `Args:` here.

chriswmackey commented on honeybee_radiance_folder/folderutil.py:
```diff
@@ -250,23 +256,31 @@ def from_dict(cls, input_dict):
+        return cls(identifier, states)
+
+    def validate(self, folder, bsdf_folder):
+        """Validate aperture group."""
```
Some `Args:` would be good here.

chriswmackey commented on setup.py:
```diff
@@ -1,17 +1,16 @@
     author_email="info@ladybug.tools",
-    description="Honeybee Radiance folder is a Python library to read, write and validate Radiance folder structure.",
+    description="Honeybee Radiance folder is a Python library to read, write and "
+    "validate Radiance folder structure.",
```
You are missing `include_package_data=True` and (I think) the `MAIFEST.IN` file to ensure that the `.cfg` file is included in the package.

mostaphaRoudsari replied:
Yes. You are right!

mostaphaRoudsari commented on honeybee_radiance_folder/folder.py:
```diff
@@ -9,12 +9,20 @@
-from .folderutil import parse_dynamic_apertures, parse_dynamic_nonapertures
+from .folderutil import parse_aperture_groups, parse_dynamic_scene
+import honeybee_radiance_folder.config as config
+import shutil
```
Thanks! Addressed.

mostaphaRoudsari commented:
Hey @chriswmackey, Thank you for your comment. I think I addressed all of them. I left my comments about suggestions for changing the name. I changed one and added a note for the other one.
